web-common/src/features/dashboards/time-controls/super-pill/new-time-dropdown/RangePickerV2.svelte

```
@@ -0,0 +1,531 @@
+    getSmallestGrainFromISODuration,
+    GrainAliasToV1TimeGrain,
+    V1TimeGrainToAlias,
+  } from "@rilldata/web-common/lib/time/new-grains";
+  import * as Popover from "@rilldata/web-common/components/popover";
+  import type { TimeGrainOptions } from "@rilldata/web-common/lib/time/defaults";
+  import TimeRangeOptionGroup from "./TimeRangeOptionGroup.svelte";
+  import RangeDisplay from "../components/RangeDisplay.svelte";
+  import TruncationSelector from "./TruncationSelector.svelte";
+  import { overrideRillTimeRef } from "../../../url-state/time-ranges/parser";
+  import { getAbbreviationForIANA } from "@rilldata/web-common/lib/time/timezone";
+  import { builderActions, Tooltip, getAttrs } from "bits-ui";
+  import TooltipContent from "@rilldata/web-common/components/tooltip/TooltipContent.svelte";
+
+  export let timeString: string | undefined;
+  export let interval: Interval<true>;
+  export let zone: string;
+  export let showDefaultItem: boolean;
+  export let context: string;
+  export let minDate: DateTime;
+  export let maxDate: DateTime;
+  export let watermark: DateTime | undefined;
+  export let smallestTimeGrain: V1TimeGrain | undefined;
+  export let defaultTimeRange: NamedRange | ISODurationString | undefined;
+  export let allowCustomTimeRange = true;
+  export let availableTimeZones: string[];
+  export let lockTimeZone = false;
+  export let onSelectTimeZone: (timeZone: string) => void;
+  export let onSelectRange: (range: string) => void;
+  export let onTimeGrainSelect: (grain: V1TimeGrain) => void;
+
+  let firstVisibleMonth: DateTime<true> = interval.start;
+  let open = false;
+  let allTimeAllowed = true;
+  let searchComponent: TimeRangeSearch;
+  let filter = "";
+  let parsedTime: RillTime | undefined = undefined;
+  let showCustomSelector = false;
+  let truncationGrain: V1TimeGrain | undefined = undefined;
+
+  $: if (timeString) {
+    try {
+      parsedTime = parseRillTime(timeString);
+    } catch {
+      // no op
```

consider adding logging instead of silently fail ?